lib/src/configuration.rs

```diff
@@ -0,0 +1,53 @@
+#[derive(Debug, Clone)]
+pub struct Configuration {
+    pub base_path: String,
+    pub user_agent: Option<String>,
+    pub basic_auth: Option<BasicAuth>,
+    pub oauth_access_token: Option<String>,
+    pub bearer_access_token: Option<String>,
+    pub api_key: Option<ApiKey>,
+    // TODO: take an oauth2 token source, similar to the go one
+}
+
+pub type BasicAuth = (String, Option<String>);
+
+#[derive(Debug, Clone)]
+pub struct ApiKey {
+    pub prefix: Option<String>,
+    pub key: String,
+}
+
+impl Configuration {
```

Comment:
Don't really know Rust, but is it normal to give a "class" (impl) and "model" (struct) the same name?

Reply:
@uniqueg Yes, this is exactly the way of doing this – we define struct first, which is about data structure fundamentally, then we implement ("impl") methods for this

Reply:
Yeah, exactly. I took reference from here: https://users.rust-lang.org/t/confusion-struct-impl-self-trait/3941/4